Makefile

```diff
@@ -209,6 +209,33 @@ check-kuttl:
 	${PGO_KUBE_CLIENT} ${KUTTL_TEST} \
 		--config testing/kuttl/kuttl-test.yaml
 
+# Set reasonable default for KUTTL_ env vars for templating
+KUTTL_PG_VERSION ?= 14
+KUTTL_PG_UPGRADE_FROM_VERSION ?= 13
+KUTTL_PSQL_CLIENT_IMAGE ?= registry.developers.crunchydata.com/crunchydata/crunchy-postgres:centos8-14.1-0
+
+# TODO: When I attempted to loop through the directories and find the yaml files,
+# due to how makefile expands variables, each file would end up in each directory.
+# So this implementation loops through to create the folders under generated
+# and then loops through all the yaml files, using a `sed` substition to place them
+.PHONY: generate-kuttl
+generate-kuttl:
+	for SRCDIR in $(shell ls testing/kuttl/source); do \
+		mkdir -p testing/kuttl/generated/$${SRCDIR}; \
+	done;
+	for FILEPATH in $(wildcard testing/kuttl/source/*/*.yaml); do \
+		NEW_FILEPATH=`echo $${FILEPATH} | sed 's/source/generated/'`; \
+		KUTTL_PG_VERSION=$(KUTTL_PG_VERSION) \
+			KUTTL_PG_UPGRADE_FROM_VERSION=$(KUTTL_PG_UPGRADE_FROM_VERSION) \
+			KUTTL_PSQL_CLIENT_IMAGE=$(KUTTL_PSQL_IMAGE) \
+			envsubst < $${FILEPATH} > $${NEW_FILEPATH}; \
+	done
+
+.PHONY: check-generated-kuttl
+check-generated-kuttl: generate-kuttl
```

Comment:
Yeah, we could have check-kuttl call generate-kuttl and run the tests both in `e2e` and in `generated`. (Eventually, if we go forward with this, we probably want to alter and move all the existing tests to `/source` and only check `generated`, yeah?)

For now I was just keeping them separate during the transition, so that we would be sure of what tests we were running.

Reply:
Awesome, I saw a mix of the two and wasn't sure what was going on. Transition does make sense, thanks for clarifying.